tests/solana_tests/primitives.rs

```diff
@@ -599,6 +604,363 @@ fn truncate_uint(n: &mut U256, width: usize) {
     }
 }
 
+#[test]
+fn test_power_overflow_boundaries() {
+    for width in (72..=256).step_by(8) {
+        let src = r#"
+        contract test {
+            function pow(uintN a, uintN b) public returns (uintN) { 
+                return a ** b;
+            }
+        }"#
+        .replace("intN", &format!("int{}", width));
+
+        let mut contract = build_solidity(&src);
+        contract.constructor("test", &[]);
+
+        let return_value = contract.function(
+            "pow",
+            &[
+                ethabi::Token::Uint(U256::from(2)),
+                ethabi::Token::Uint(U256::from(width - 1)),
+            ],
+            &[],
+            None,
+        );
+
+        let res = BigUint::from(2_u32).pow((width - 1) as u32);
+
+        assert_eq!(
+            return_value,
+            vec![ethabi::Token::Uint(U256::from_big_endian(
+                &res.to_bytes_be()
+            ))]
+        );
+
+        let sesa = contract.function_must_fail(
+            "pow",
+            &[
+                ethabi::Token::Uint(U256::from(2)),
+                ethabi::Token::Uint(U256::from(width)),
+            ],
+            &[],
+            None,
+        );
+
+        assert_ne!(sesa, Ok(0));
+    }
+}
+
+#[test]
+fn test_overflow_boundaries() {
+    // For bit sizes from 8..64, LLVM has intrinsic functions for multiplication with overflow. Testing starts from int types of 72 and up. There is no need to test intrinsic LLVM functions.
```

Comment:
`There is no need to test intrinsic LLVM functions.`
If you have used them, you need to check at least that they are correctly wired up.

Reply:
Call them for one case to check if the code around them is working correctly.

Reply:
I have not used them.
They are called for bit sizes 8..64.

Reply:
Ah, I see.